### PR TITLE
FEAT: dlloader class

### PR DESCRIPTION
**Pull request by mathieurobert1**
I created the DLLoader to load components and systems .so files and added the folder Server/lib to store components and systems.

I fix some issues for SparseArray and Register to store shared_pointers.
Finally, I implemented the IComponent and ISystem classes to handle components and systems with the DLLoader.

Here the tutorials to create components : https://slimwiki.com/6fvu0f5pu/getting-started-nqdxve9qg-/ecs-component-w9on3dn-o-mi3jl3gpofei
And to create systems: https://slimwiki.com/6fvu0f5pu/getting-started-nqdxve9qg-/ecs-system-oyzc-3tlv-toxcperq4hri

More details : #20

### CODE REVIEW

**Review by Peralban — CHANGES_REQUESTED** [review]
Delete this include folder, create an interface folder for interface and put cpp and hpp in src.

File: Ecs/src/SparseArray.hpp [hunk 1]
```diff
@@ -7,143 +7,109 @@
...
-         *
-         * @param other SparseArray to move.
-         * @return SparseArray& this
          */
-        SparseArray & operator=( SparseArray &&other) noexcept
-        {
-            _data = other._data;
-            return *this;
-        };
+        SparseArray& operator=(SparseArray &&other) noexcept = default;
```
Comment (Thomaltarix) [inline Ecs/src/SparseArray.hpp:92]: Same

File: Ecs/src/SparseArray.hpp [hunk 2]
```diff
@@ -212,94 +173,76 @@ class SparseArray {
...
+         * @brief Erases the element at the specified index.
          */
-        void erase(size_type pos)
-        {
-            _data.erase(_data.begin() + pos);
-        };
+        void erase(size_type pos) {
+            if (pos < _data.size()) {
+                _data.erase(_data.begin() + pos);
+            }
```
Comment (Thomaltarix) [inline Ecs/src/SparseArray.hpp:226]: Useless brackets

File: Ecs/src/SparseArray.hpp [hunk 3]
```diff
@@ -212,94 +173,76 @@ class SparseArray {
...
+         * @tparam Params Parameters to pass to the component's constructor.
+         * @return reference_type Reference to the new component.
          */
         template <class... Params>
-        reference_type emplace_at(size_type pos, Params&&... params)
-        {
-            if (pos >= _data.size())
+        reference_type emplace_at(size_type pos, Params&&... params) {
+            if (pos >= _data.size()) {
                 _data.resize(pos + 1);
```
Comment (Thomaltarix) [inline Ecs/src/SparseArray.hpp:214]: Useless brackets

File: Ecs/src/SparseArray.hpp [hunk 4]
```diff
@@ -212,94 +173,76 @@ class SparseArray {
...
+         * @return size_type Index of the component, or -1 if not found.
          */
-        size_type get_index(value_type const &value) const
-        {
+        size_type get_index(const value_type& value) const {
             for (size_type i = 0; i < _data.size(); ++i) {
-                if (_data[i] == value)
+                if (_data[i] == value) {
                     return i;
+                }
```
Comment (Thomaltarix) [inline Ecs/src/SparseArray.hpp:239]: Useless brackets

File: Server/include/IComponent.hpp [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+** File description:
+** IComponent
+*/
+
+#pragma once
+
+#include <string>
+
+/**
+ * @brief
```
Comment (Thomaltarix) [inline Server/include/IComponent.hpp:13]: Nice brief

File: Server/include/ISystem.hpp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+#include "../../Ecs/src/Registry.hpp"
+
+/**
+ * @brief Interface class for ECS systems.
+ *
+ */
+class ISystem {
+
+    public:
+
```
Comment (Thomaltarix) [inline Server/include/ISystem.hpp:22]: A destructor is missing

**Review by Thomaltarix — CHANGES_REQUESTED** [review]

**Review by theophile-jr — DISMISSED** [review]

**Review by AxelF44 — CHANGES_REQUESTED** [review]

**Review by Peralban — DISMISSED** [review]

**Review by Thomaltarix — DISMISSED** [review]

**Review by Peralban — APPROVED** [review]

**Review by theophile-jr — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

**Review by Thomaltarix — APPROVED** [review]

**Pull request merged**